src/GafferScene/LightFilter.cpp

```diff
@@ -137,11 +149,17 @@ IECore::ConstObjectPtr LightFilter::computeSource( const Context *context ) cons
 void LightFilter::hashAttributes( const SceneNode::ScenePath &path, const Gaffer::Context *context, const ScenePlug *parent, IECore::MurmurHash &h ) const
 {
 	h.append( shaderPlug()->attributesHash() );
+	filteredLightsPlug()->hash( h );
 }
 
 IECore::ConstCompoundObjectPtr LightFilter::computeAttributes( const SceneNode::ScenePath &path, const Gaffer::Context *context, const ScenePlug *parent ) const
 {
-	return shaderPlug()->attributes();
+	IECore::CompoundObjectPtr result = new IECore::CompoundObject;
+
+	result->members() = shaderPlug()->attributes()->members();
+	result->members()[g_filteredLightsAttributeName] = new IECore::StringData( filteredLightsPlug()->getValue() ); 
```

Comment:
Is this definitely what we want when the string is empty? Perhaps it's better to not make the attribute in that case, so that several default-valued blockers could be grouped, and their linking set once with a StandardAttributes at the group level?

Reply:
Good point - 237d6e0. I've also extended the test in 8803536.